### PR TITLE
MultiQC reports 

### PR DESCRIPTION
**Pull request by kate-v-stepanova**
Displaying MultiQC reports on project page and bioinfo tab.
On the project page it is displayed as a tab. Additionally, on both project page and bioinfo tab, there is link to open it as a new page.
If no multiqc reports available for current project, the link is not displayed.

Examples:
1. Project page, multiqc report available: https://genomics-status-stage.scilifelab.se/project/P4601
2. Project page, multiqc report not available: https://genomics-status-stage.scilifelab.se/project/P4605
3. Bioinfo tab, multiqc report available: https://genomics-status-stage.scilifelab.se/bioinfo/P4601
4. Bioinfo tab, multiqc report not available: https://genomics-status-stage.scilifelab.se/bioinfo/P4605

Also, one thing about implementation. Currently to check if multiqc reports exist, I have a method `check_multiqc` in both ProjectSamplesHandler, and BioinfoTabHandler. Which I don't think is very good, because the same code is repeated in two different places. 
Instead, I could implement this method in `BaseHandler`, as both these classes are inherited from it. 
What do you think?

### CODE REVIEW

**Comment by vezzi** [comment]
HI @kate-v-stepanova when following one of the links I get "Bad Gateway, The proxy server received an invalid response from an upstream server."
genomics-status-stage seems to be up but extremely slow...

**Comment by ewels** [comment]
I had the same problem last night. Seems to be working now but slow as you say.

Changes look great anyway! Few minor points:

* Is it worth using a different endpoint for the reports that isn't `/api/`? As it's not a JSON result so not really an api call
* I prefer putting the button on the bioinfo tab in line with the title, with `.pull-right` so that it floats on the right hand side of the screen.
* Personally, I'd rather just have no tab at all if there's no MultiQC report. Instead of a tab and an alert.
* The `<iframe>` resizing thing you've done is nice, but it's going to cause problems. For me it's not tall enough and cuts off the bottom of the report. Also it doesn't update when the window is resized (narrower = taller page, more report cut off).
  * At the very least, it needs to have `overflow: auto` set so that it's possible to scroll within the iframe
  * Personally, I think it's better to ditch this `iframe` idea of mine and just stick with buttons that open the report in a new browser tab.
  * Other problem is that MultiQC reports are big and can crash the browser tab sometimes. Better to isolate in their own tab with their own memory resources.

I'll comment on the code specific as well, these are general points. Looking great though, good work!

File: status/bioinfo_analysis.py [hunk 1]
```diff
@@ -185,8 +188,26 @@ def get(self, project_id):
...
+        # get only the first one
+        for row in rows:
+            project_name = row.value.get('project_name', '')
+            break
+
+        if project_name:
+            multiqc_name = '{}_multiqc_report.html'.format(project_name)
+            multiqc_path = self.application.multiqc_path or ''
+            multiqc_path = os.path.join(multiqc_path, multiqc_name)
+            print multiqc_path
```
Comment (ewels) [inline status/bioinfo_analysis.py]: Is this left over from testing?

File: status/multiqc_report.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                with open(multiqc_path, 'r') as multiqc_file:
+                    html = multiqc_file.read()
+                    return html
+
+    def get(self, project):
+        multiqc_report = self.get_multiqc(project)
+        if multiqc_report:
+            self.write(multiqc_report)
+        else:
+            self.write("<div class='alert alert-warning'>No MultiQC reports available</div>")
```
Comment (ewels) [inline status/multiqc_report.py]: Needs to be within the genomics status page, currently this is just a text string without any other HTML.

Something like this should work I guess?

```python
t = self.application.loader.load("error_page.html")
self.write(t.generate(gs_globals=self.application.gs_globals, error="404", reason="MultiQC Report Not Found", user=self.get_current_user_name()))
```

File: status_app.py [hunk 3]
```diff
@@ -133,6 +134,7 @@ def __init__(self, settings):
             ("/api/v1/last_psul", LastPSULRunHandler),
             ("/api/v1/lab_status/([^/]*)", ProjectLabStatusHandler),
             ("/api/v1/load_workset_samples", WorksetSampleLoadHandler),
+            ("/api/v1/multiqc_report/([^/]*)$", MultiQCReportHandler),
```
Comment (ewels) [inline status_app.py]: I think just `/multiqc_report/` instead of `/api/v1/multiqc_report/`

**Comment by ewels** [comment]
Code looks good! As you mentioned, the report checking code is duplicated which we want to avoid. So yes, I would add it as a new function in `BaseHandler`.

**Comment by kate-v-stepanova** [comment]
All requested changes done:
1. Replace MultiQC tab on project page with a button
2. Moved button on bioinfo tab to the right
3. Moved `get_multiqc` method to `BaseHandler`
4. Changed url to `/multiqc_reports/project_id`
5. Error page is displayed when no multiqc reports are available

**Comment by kate-v-stepanova** [comment]
Yes, and **IMPORTANT THING**: we need to copy multiqc reports from irma to `tools-dev` as well. For testing purpose, I copied a couple of files manually, and pointed genomics-status to this location. But reports are not copied automatically to `/var/local/mqc_reports/`, as it is done on `tools`

**Comment by ewels** [comment]
Great, looks good! 👍

**Comment by ewels** [comment]
..deployed

**Pull request merged**